lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -268,14 +277,24 @@ newStakePoolLayer db@DBLayer{..} nl tr = StakePoolLayer
 
 readPoolsPerformances
     :: DBLayer m
-    -> EpochNo
+    -> EpochLength
+    -> SlotId
     -> m (Map PoolId Double)
-readPoolsPerformances DBLayer{..} (EpochNo epochNo) = do
-    let range = [max 0 (fromIntegral epochNo - 14) .. fromIntegral epochNo]
+readPoolsPerformances DBLayer{..} (EpochLength el) tip = do
+    let range = [max 0 (currentEpoch - 14) .. currentEpoch]
     atomically $ fmap avg $ forM range $ \ep -> calculatePerformance
+        (slotsInEpoch ep)
         <$> (Map.fromList <$> readStakeDistribution ep)
         <*> (count <$> readPoolProduction ep)
   where
+    currentEpoch = tip ^. #epochNumber
+
+    slotsInEpoch :: EpochNo -> Int
```

Comment:
Should maybe be `Quantity "slot" Word64` or `SlotNo`

Reply:
That would be true for a function that is part of the module public API, here, the function is a local declaration and it has less value IMO

Reply:
It _is_ actually part of the public API as the argument of `calculatePerformance`. Would you be fine with… maybe `SlotNo`?

Using `SlotNo` here is a _slight_ stretch as we would use it to represent the "number of slots" instead of "the slot number of a slot", which are very closely related but arguably not exactly the same.

I'd go with it though.

Reply:
I think `Int` is fine to be honest.